### PR TITLE
Add the ability to restore the mentor to the course

### LINKED ISSUE #861: [Course Mentors] Возможность восстановить отчисленного ментора
Возможность восстановить отчисленного ментора на странице [Course Mentors](https://app.rs.school/course/admin/mentors?course=js-fe-2021Q3)

image

### PR DESCRIPTION
**Pull request by Pulya10c**
fixes  #861

### CODE REVIEW

File: client/src/pages/course/admin/mentors.tsx [hunk 1]
```diff
@@ -75,6 +75,19 @@ function Page(props: CoursePageProps) {
     }
   };
 
+  const handleRestore = async ({ githubId }: MentorDetails) => {
+    try {
+      setLoading(true);
+      await service.restoreMentor(githubId);
+      const records = mentors.map(r => (r.githubId === githubId ? { ...r, isActive: true } : r));
+      setMentors(records);
```
Comment (mikhama) [inline client/src/pages/course/admin/mentors.tsx:83]: You cannot rely on "mentors" state.
Please the next form: `setMentors(prevState => /* do something with prevState */)`.
https://reactjs.org/docs/hooks-reference.html#usestate

Comment (Pulya10c) [reply client/src/pages/course/admin/mentors.tsx]: Done

**Review by apalchys — APPROVED** [review]

**Review by apalchys — CHANGES_REQUESTED** [review]
Please move restoreMentor method to `rsschool-app/server/src/services/mentor.service.ts`

File: server/src/services/course.service.ts [hunk 2]
```diff
@@ -200,6 +200,13 @@ export async function expelMentor(courseId: number, githubId: string) {
   }
 }
 
+export async function restoreMentor(courseId: number, githubId: string) {
```
Comment (apalchys) [inline server/src/services/course.service.ts:203]: please move to `rsschool-app/server/src/services/mentor.service.ts`

Comment (Pulya10c) [reply server/src/services/course.service.ts:203]: Why should this be taken out of the course - this is the same expulsion and restoration of the mentor in the current course?

**Pull request merged**